docs/getting-started.md

```diff
@@ -144,6 +144,27 @@ Experiment tracker
 
 By default, Matcha will hide sensitive resource properties. If you need one of these properties, then you can add the `--show-sensitive` flag to your `get` command.
 
+# &#129309; Sharing Resources
+
```

Comment:
Could we add a piece on how it works as an introduction, rather than just how to use shared state. Something along the lines of "When Matcha provision first runs, it creates a storage blob in Azure which holds details of the provisioned environment. For more detail, please see our `Inside Matcha > How does Matcha work` section [linked]"